### DISCUSSION TITLE
Should I expect to see all of my control plane nodes setting the VIP?

### DISCUSSION THREAD

**Discussion by v1nsai** [siderolabs/talos#9074] [Q&A] [answered]
This question may reveal my ignorance of how a virtual IP works, but I noticed that when I installed my first control plane node and included a VIP, it added it to the list of IP addresses in the upper right corner of the dashboard.  When I installed my second control plane node with a VIP, it did not add it to its list of IPs.

After adding/removing the VIP from the second node and trying to get it to claim it, it occurred to me that this may not be expected behavior.  Can anyone confirm/deny that I should be seeing both nodes claim the VIP before I waste any more time?  Thanks!

**Comment by frezbo** [comment, marked as answer]
VIP is only active on one controlplane at a time, it resides with the controlplane that's the etcd leader

**Comment by smira** [comment]
Also please keep in mind that running _two_ nodes etcd cluster doesn't make much sense, as if one the nodes fails, etcd would lose quorum. VIP and multi-node controlplane make sense only with 3 (or 5, 7, ...) nodes.

**Comment by The-nuT** [comment]
I ran into this as well, after a day of trial and error here is what I found.

ENV:
3 proxmox nodes, running as CP and Worker nodes on all 3 with static IP's.
 
https://X.X.3.35:6443 is the CP API ingress point.
3.36 .37 and .38 are my 3 nodes

talosconfig file needs to have to have all 3 nodes in it for endpoints.  I.E.

context: talosclupm-1
contexts:
    talosclupm-1:
        endpoints:
            - X.X.3.36
            - X.X.3.37
            - X.X.3.38
        ca: 12345abcde==
        crt:  a;kldjl;kajl;kjdeavaerewsdf3
        key: 12oijl4k2oiduafoihjdf

You also need to make sure that your machine certCAs have your IP listed in the machine config and or controlplane.yaml file for each node config file. I.E.

    # Extra certificate subject alternative names for the machine's certificate.
    certSANs:
    #   # Uncomment this to enable SANs.
    #   - 10.0.0.10
    #   - 172.16.0.10
    #   - 192.168.0.10
         **- X.X.3.37** on 37
         **- X.X.3.38** on 38  etc...     

    # Used to provide additional options to the kubelet.
    kubelet:
        image: ghcr.io/siderolabs/kubelet:v1.32.0 # The `image` field is an optional reference to an alternative kubelet image.
        # The `extraArgs` field is used to provide additional flags to the kubelet.

Check that the VIP is showing up on one of the nodes

talosctl -n <IP> get addresses | grep <IP>

SIDE NOTE:
I also needed to change my interfaces from ens18 (the VMs label) to eth0 in the config file and made sure net.ifnames=0 was set in the machine config file(s)
